app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,74 @@
+class OrdersController < ApplicationController
+  before_action :logged_in_user
+
+  def new
+    @cart_items = load_product_in_cart
+    @total_price_cart = total_price_cart @cart_items
+    @addresses = current_user.addresses.address_order
+  end
+
+  def create
+    ActiveRecord::Base.transaction do
+      build_order
+      @result = add_order_detail @order
+      @order.save! if @result
```

Comment:
```suggestion
      add_order_detail @order
      @order.save!
```
nên ri bé nhé, nếu có invalid thì sẽ raiise exception để còn rollabck address nữa chi bé

Reply:
Chị ơi chỗ này em trả về false chớ em chưa raiise exception